brave/brave-core
# Binance version 2 trading widget

Fixes: https://github.com/brave/brave-browser/issues/9118

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
Pre-requisite: Ensure you are signed up for Binance.com

1. Ensure that the functionality of the Buy widget is maintained as per existing behavior. This also includes checking the show/hide widget menu functionality for regressions.
2. Ensure that for .com users, clicking `Connect` via the Buy screen leads you to the Oauth flow for Binance.com, where you can authorize the Brave NTP widget. 
3. Upon being redirected back to the NTP, you should see the full widget realize after being authed, starting on the deposit screen.

### Deposit View:
1. Ensure that assets filter and can be searched via the search input.
2. Ensure that each asset can be clicked in to for a detail view.
3. For addresses that are available, ensure that a "Copy" button is present, upon which clicking will copy the asset address to your clipboard. The QR code icon should be clickable to render a full widget window. Upon scanning this QR code it should render as the asset address. Exiting out of the screen should lead back to the deposit view.
4. For unavailable addresses, neither the QR icon or copy button should be present.

### Buy View:
This is expected to be the same as the initial un-authed Buy screen, just contained within the view.

### Convert View
1. Ensure that changing the asset you wish to convert updates the available balance displayed for that asset.
2. Ensure that the available "Convert To" options change along with that asset (not every asset can be converted in to another.)
3. Ensure that you can not enter amount an amount greater than the balance of the convert from asset ("Upon hitting preview conversion you should receive an error")
**Note, very small amounts cannot be accepted for conversion, please at leas try $5 of ETH or the source asset**
4. Ensure that when entering an acceptable <= amount, you are greeted with a conversion preview screen. With a counter down from 30 seconds.
5. Ensure that after thirty seconds have passed, that the quote expires and the conversion preview screen goes away.
6. Ensure that trying again produces another preview screen, and that clicking on "confirm" leads to a successful conversion screen.
7. Refresh intervals are every 30 seconds for balance, but conversion results take a few minutes. Ensure the widget conversion balance values are ultimately reflected with what displays in your binance.com dashboard.

### Summary view:
1. The BTC asset should be the first and most prominent asset, with the account value in Bitcoin and subsequent value in USD. In smaller, less prominent form, all respective assets with their totals and USD values should be listed.
2. Clicking the "eye" to the right of the BTC item should obscure/unobscure all balances/values. This should persist upon reload.

### Screenshots:
<img width="310" alt="Screen Shot 2020-04-05 at 11 53 11 PM" src="https://user-images.githubusercontent.com/8732757/78530896-5354f600-7799-11ea-9e2c-a1fdaeb40ded.png">
<img width="321" alt="Screen Shot 2020-04-05 at 11 53 06 PM" src="https://user-images.githubusercontent.com/8732757/78530897-54862300-7799-11ea-8cc0-519985e476ec.png">
<img width="320" alt="Screen Shot 2020-04-05 at 11 52 53 PM" src="https://user-images.githubusercontent.com/8732757/78530900-551eb980-7799-11ea-9ac2-7e21a8e8e199.png">
<img width="317" alt="Screen Shot 2020-04-05 at 11 52 40 PM" src="https://user-images.githubusercontent.com/8732757/78530903-551eb980-7799-11ea-9eba-cc40ce9b7fa4.png">
<img width="328" alt="Screen Shot 2020-04-05 at 11 52 35 PM" src="https://user-images.githubusercontent.com/8732757/78530904-55b75000-7799-11ea-886f-fe0599788012.png">
<img width="315" alt="Screen Shot 2020-04-05 at 11 52 31 PM" src="https://user-images.githubusercontent.com/8732757/78530909-564fe680-7799-11ea-8c67-e20e8466a4e9.png">


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
